client/landing/stepper/declarative-flow/import-flow.ts

```diff
@@ -219,13 +220,23 @@ const importFlow: Flow = {
 		};
 
 		const goBack = () => {
+			const source = urlQueryParams.get( 'source' );
+
 			switch ( _currentStep ) {
+				case 'import': {
+					if ( source === 'sites-dashboard' ) {
+						return window.location.assign( '/sites' );
+					}
+
+					return;
+				}
+
 				case 'importList':
 					// eslint-disable-next-line no-case-declarations
 					const backToStep = urlQueryParams.get( 'backToStep' );
 
 					if ( backToStep ) {
-						const path = `${ backToStep }?siteSlug=${ siteSlugParam }`;
+						const path = addQueryArgs( backToStep, { siteSlug: siteSlugParam, source } );
```

Comment:
By default, the `navigate()` function retains all query parameters as you move between steps. However as soon as you modify the query parameters you're expected to manage all of them yourself.

https://github.com/Automattic/wp-calypso/blob/57df03537af291e9775a93cb293cddfdc03bd55c/client/landing/stepper/declarative-flow/internals/index.tsx#L92

So we need to manually pass the `?source` param manually.